modules/programs/vim.nix

```diff
@@ -65,6 +65,18 @@ in {
     programs.vim = {
       enable = mkEnableOption "Vim";
 
+      sourcePackage = mkOption {
+        type = with types; package;
+        default = pkgs.vim_configurable;
+        example = literalExample
+          "pkgs.vim_configurable.override { python = pkgs.python3; }";
+        description = ''
+          A Vim package which will be installed and configured by Home Manager.
+          This can either be one of the default Vim packages from Nixpkgs or a customized
+          package.
+        '';
```

Comment:
Considering the `customize` logic at the bottom, I see two options:
1. Document the need for the package to have a `customize` function.
2. Use `(pkgs.vimUtils.makeCustomizable cfg.package).customize` on the package.
   @teto what would happen if `pkgs.vimUtils.makeCustomizable` were to be called on a customizable package? double-wrapping?

Reply:
Looking at the code: https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/vim-plugins/vim-utils.nix#L396-L418

If my Nix-fu doesn't fail me, it doesn't seem it double wraps. The behavior may still be somewhat surprising since this module customization will probably overwrite the package passed via `package` option.

I still think the assertion seems to be a better solution, but @teto probably knows much better about vim than me.